sdk/core/azure-core/inc/azure/core/http/curl_transport.hpp

```diff
@@ -16,9 +16,6 @@
 
 #if defined(AZ_PLATFORM_LINUX)
 #include <openssl/opensslv.h>
-#if OPENSSL_VERSION_NUMBER >= 0x00905100L
-#define _azure_SUPPORT_SETTING_CAPATH
-#endif // OPENSSL_VERSION_NUMBER >= 0x00905100L
 #endif // defined(AZ_PLATFORM_LINUX)
 
```

Comment:
`#include <openssl/opensslv.h>` is no longer needed - the whole reason we added it was to make sure that the `OPENSSL_VERSION_NUMBER` is defined. Please drop this entire `ifdef` section, as it will be empty after we remove the `include` (I would've posted a GH suggestion comment, but GitHub is not letting me to do it).

Reply:
I'm not comfortable removing the azure_SUPPORT_SETTING_CAPATH option and replacing with with AZ_PLATFORM_LINUX.

The reason that the original submitter used SUPPORT_SETTING_CAPATH was to enable a single location which determines whether the CAPATH option should be exposed. That logic could be potentially complicated, since the CAPath option is only available for openssl, mbedtls and one other crypto backend.

Reply:
I'd rather complicate it and introduce `_azure_SUPPORT_SETTING_CAPATH` when and if we have more complex condition.
It is internal and seems like more of an internal thing to me, so either way is fine, but I'd prefer the simple platform condition.

Reply:
We should support what we test and base it on customer need, while keeping complexity low.

If we want to support other backends, and fragment the API surface (which I really vote to  minimize if not avoid), we'd need to test as well.

Reply:
The problem here is that CURL supports multiple backends. And the customer gets to configure vcpkg to use whichever CURL backend that they choose. We don't control that customer choice. All we can say is that we want curl. But the customer gets to pick the flavor of curl.

By taking a dependency on CURL, we have accepted support of other backends, because that's what CURL supports.

Reply:
Since this is planned for beta, I would love to get customer feedback on this, and keep the conditional include simple to start with. Having customers hit a non-supporte backend would be good feedback, tbh.